lib/db_utils.go

```diff
@@ -529,7 +529,28 @@ type DBPrefixes struct {
 	// Prefix -> <RandomSeedHash [32]byte>.
 	PrefixCurrentRandomSeedHash []byte `prefix_id:"[84]" is_state:"true"`
 
-	// NEXT_TAG: 85
+	// PrefixSnapshotGlobalParamsEntry: Retrieve a snapshot GlobalParamsEntry by SnapshotAtEpochNumber.
+	// Prefix, SnapshotAtEpochNumber -> *GlobalParamsEntry
+	PrefixSnapshotGlobalParamsEntry []byte `prefix_id:"[85]" is_state:"true"`
+
+	// PrefixSnapshotValidatorByPKID: Retrieve a snapshot ValidatorEntry by <SnapshotAtEpochNumber, PKID>.
+	// Prefix, SnapshotAtEpochNumber, ValidatorPKID -> *ValidatorEntry
+	PrefixSnapshotValidatorByPKID []byte `prefix_id:"[86]" is_state:"true"`
+
+	// PrefixSnapshotValidatorByStatusAndStake: Retrieve stake-ordered active ValidatorEntries by SnapshotAtEpochNumber.
+	// Prefix, SnapshotAtEpochNumber, Status, TotalStakeAmountNanos, ValidatorPKID -> nil
+	// Note: we parse the ValidatorPKID from the key and the value is nil to save space.
+	PrefixSnapshotValidatorByStatusAndStake []byte `prefix_id:"[87]" is_state:"true"`
+
+	// PrefixSnapshotGlobalActiveStakeAmountNanos: Retrieve a snapshot GlobalActiveStakeAmountNanos by SnapshotAtEpochNumber.
+	// Prefix, SnapshotAtEpochNumber -> *uint256.Int
+	PrefixSnapshotGlobalActiveStakeAmountNanos []byte `prefix_id:"[88]" is_state:"true"`
+
+	// PrefixSnapshotLeaderSchedule: Retrieve a ValidatorPKID by <SnapshotAtEpochNumber, LeaderIndex>.
+	// Prefix, SnapshotAtEpochNumber, LeaderIndex -> ValidatorPKID
+	PrefixSnapshotLeaderSchedule []byte `prefix_id:"[89]" is_state:"true"`
```

Comment:
The ValidatorPKID is not included in the key. This is deliberate. We want to be able to say, who is the first validator in the leader schedule that was snapshot in epoch = N. `SnapshotAtEpochNumber = N, LeaderIndex = 0`. This way we are guaranteed there will every be one such row, with the ValidatorPKID being the value. If we included the ValidatorPKID in the key, we would either have to query with the ValidatorPKID (already knowing which validator is there which is kinda a catch-22) or there could be a weird case where there are multiple validators assigned to the same index within the same epoch since multiple keys can exist for the same SnapshotAtEpochNumber + LeaderIndex.